src/neptune/github.d

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+
+    Github interaction helpers
+
+    Copyright:
+        Copyright (c) 2017 Sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module neptune.github;
+
+import neptune.api;
+import octod.api.repos;
+
+/// Structure defining a name-tag association
+struct Head
+{
+    string name;
+    string sha;
+}
+
+/*******************************************************************************
+
+    Params:
+        repo = repository object
+
+    Returns:
+        all branches for this repository
+
+*******************************************************************************/
+
+public Head[] getBranches ( ref Repository repo )
+{
+    import std.format;
+    import std.algorithm;
+    import std.range;
+
+    import vibe.data.json;
+
+    auto owner = repo.json["owner"]["login"].get!string();
```

Comment:
Every function that does some json access may fail if github responds with something entirely different. You may want some error handling for that.